esp-wifi/src/lib.rs

```diff
@@ -134,8 +134,13 @@ mod binary {
 }
 mod compat;
 
+#[cfg(not(feature = "preempt-extern"))]
 mod preempt;
 
+#[cfg(feature = "preempt-extern")]
+#[path = "preempt_extern.rs"]
+pub mod preempt;
```

Comment:
This needs to be public so `preempt::Scheduler` is available.

Reply:
At this point this should probably be always on & public, and our internal implementation can register a Scheduler as a third party one would. We could gate the builtin behind a `builtin-scheduler` default-enabled feature, that way we would conform to cargo's "features should be additive" idea.

Reply:
Implemented in [c4024321](https://github.com/esp-rs/esp-hal/pull/3115/commits/c40243216d343532c6ae21f4d5ceb09ea28566c3).

(It does feel a bit odd to go through the opaque raw pointers within the same crate.)

Reply:
We probably (read: I want to :D) move the scheduler bits out of here anyways, so I think keeping us honest is the right call here.